### PR TITLE
add debug doc for rudr

### LINKED ISSUE #418: docs: debugging doc?
There currently isn't a debugging article, which could be very helpful for "known issues" or "common blunders." For example, my component failed to deployed and I don't know where to go from here.

### PR DESCRIPTION
**Pull request by wonderflow**
Now we don't have very good ways to debug, so I try to conclude some of my own way.

We could add more into this debug doc in the near future.

 fixes #418

### CODE REVIEW

File: docs/developer/debug.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# How to Debug?
+
+Now Rudr is still in heavily develop, if you have troubles playing with it.
```
Comment (suhuruli) [inline docs/developer/debug.md]:
```suggestion
```

File: docs/developer/debug.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+You could try to debug with following steps.
```
Comment (suhuruli) [inline docs/developer/debug.md]:
```suggestion
This document outlines troubleshooting guides to help you debug things when deployments via Rudr are not successful.```

File: docs/developer/debug.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Check Rudr pod is running
+
+Make sure rudr is running successfully.
```
Comment (suhuruli) [inline docs/developer/debug.md]: We should be consisting with the casing for Rudr.

File: docs/developer/debug.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+## Check Rudr logs
+
+When you deploy an Application Configuration, you could find information about it from logs.
+
+```
+$ kubectl logs -f rudr-6b9b9c57cd-zxgfr 
+```
+
+Currently, all workload implemented in Rudr will create pods.
+So if your Application Configuration didn't create any pod, there must be some error information in the logs.
```
Comment (suhuruli) [inline docs/developer/debug.md]:
```suggestion
So if your ApplicationConfiguration didn't create any pod, there must be some error information in the logs.
```

File: docs/developer/debug.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+Check the logs and report to us.
```
Comment (suhuruli) [inline docs/developer/debug.md]:
```suggestion
Check the logs and file an issue with the log information. 
```

File: docs/developer/debug.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+## Check Component exist
```
Comment (suhuruli) [inline docs/developer/debug.md]:
```suggestion
## Check if the Component exists
```

File: docs/developer/debug.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+hpa-example-replicated-v1    3s
+nginx-replicated-v1          3s
+nginx-singleton-v1           3s
+```
+
+Don't forget to deploy component before using them.
+
+## Check parameters
+
+Check the parameters and properties of Application Configuration are consistent with trait and component.
```
Comment (suhuruli) [inline docs/developer/debug.md]:
```suggestion
Check the parameters and properties of ApplicationConfiguration are consistent with trait and component.   
```

**Comment by wonderflow** [comment]
@suhuruli I think at least this is a good start to write this debug doc.

**Comment by suhuruli** [comment]
@erikadoyle , can you also give this one a review please?

**Pull request merged**